### PR TITLE
[BugFix] When the reserved keyword uses backquotes as input, deserialization causes  parseSqlToExpr syntax error, which results in log edit failure.

### LINKED ISSUE #51678: When the reserved keyword uses backquotes as input, deserialization causes parseSqlToExpr syntax error, which results in a checkpoint failure.
### Steps to reproduce the behavior (Required)
CREATE TABLE `uds_nio_user_group_data` ( `group_code` varchar(256) NOT NULL COMMENT "verison code", `group_data_version` varchar(32) NOT NULL COMMENT "version", `entity_id` varchar(64) NOT NULL COMMENT "version id", `partition` date NOT NULL COMMENT "verison" ) ENGINE=OLAP DUPLICATE KEY(`group_code`, `group_data_version`, `entity_id`) PARTITION BY date_trunc('day',`partition`) DISTRIBUTED BY HASH(`entity_id`) BUCKETS 30 PROPERTIES ( "replication_num" = "3", "in_memory" = "false", "enable_persistent_index" = "false", "replicated_storage" = "true", "compression" = "LZ4" )

### Expected behavior (Required)
the fe state is normal
### Real behavior (Required)
```
2024-10-09 19:57:57.156+08:00 WARN (stateChangeExecutor|79) [GlobalStateMgr.replayJournalInner():2395] catch exception when replaying journal, id: 790, data: null,
com.starrocks.journal.JournalException: fail to read journal entity key=790, data=<DatabaseEntry offset="0" size="23373" data="50 200 0 0 91 71 123 34 100 34 58 34 116 101 115 116 34 44 34 116 34 58 123 34 99 108 97 122 122 34 58 34 79 108 97 112 84 97 98 108 101 34 44 34 99 108 117 115 116 101 114 73 100 34 58 49 49 50 49 51 54 54 56 51 51 44 34 115 116 97 116 101 34 58 34 78 79 82 77 65 76 34 44 34 105 110 100 101 120 73 100 84 111 77 101 116 97 34 58 123  ... 23273 bytes not shown "/>
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:254) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:292) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:2374) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:2333) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.transferToLeader(GlobalStateMgr.java:1317) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.access$100(GlobalStateMgr.java:348) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$1.transferToLeader(GlobalStateMgr.java:820) ~[starrocks-fe.jar:?]
        at com.starrocks.ha.StateChangeExecutor.runOneCycle(StateChangeExecutor.java:103) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:107) ~[starrocks-fe.jar:?]
Caused by: com.starrocks.sql.parser.ParsingException: Getting syntax error at line 1, column 10. Detail message: Unexpected input '(', the most similar input is {<EOF>}.
        at com.starrocks.sql.parser.ErrorHandler.syntaxError(ErrorHandler.java:40) ~[starrocks-fe.jar:?]
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at com.starrocks.sql.parser.StarRocksDefaultErrorStrategy.reportInputMismatch(StarRocksDefaultErrorStrategy.java:78) ~[starrocks-fe.jar:?]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at com.starrocks.sql.parser.StarRocksParser.expressionSingleton(StarRocksParser.java:39163) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.SqlParser.invokeParser(SqlParser.java:251) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.SqlParser.parseSqlToExpr(SqlParser.java:195) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.ExpressionRangePartitionInfo.gsonPostProcess(ExpressionRangePartitionInfo.java:95) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.gson.GsonUtils$ProcessHookTypeAdapterFactory$1.read(GsonUtils.java:689) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.RuntimeTypeAdapterFactory$1.read(RuntimeTypeAdapterFactory.java:415) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.GsonUtils$ProcessHookTypeAdapterFactory$1.read(GsonUtils.java:687) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.RuntimeTypeAdapterFactory$1.read(RuntimeTypeAdapterFactory.java:415) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.GsonUtils$ProcessHookTypeAdapterFactory$1.read(GsonUtils.java:687) ~[starrocks-fe.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:963) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:928) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:877) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:848) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:274) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:248) ~[starrocks-fe.jar:?]
        ... 8 more
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by zhangheihei**
## Why I'm doing:
There is a difference between that the pr and [51781](https://github.com/StarRocks/starrocks/pull/51781).
Error scenario encountered with the create sql in version3.3:
1 create tablet succeeds
2 getEditLog().logCreateTable reported error
3 Restart FE, table is missing.
image

## What I'm doing:
Fixes #51678

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by Seaven** [comment]
fix test case plz, and you can add some UT

**Comment by zhangheihei** [comment]
hi @kangkaisen @Seaven 
 tks reveiw.
https://github.com/StarRocks/starrocks/pull/45842 refactored here to use ColumnId instead of Identifier str.
I'm going to close this PR and re-pull the branch from branch-3.2.

**Review by trueeyu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/persist/ColumnIdExpr.java [hunk 1]
```diff
@@ -137,6 +137,9 @@ public String visitSlot(SlotRef node, Void context) {
             if (node.getTblNameWithoutAnalyzed() != null) {
                 return node.getTblNameWithoutAnalyzed().toString() + "." + node.getColumnId().getId();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/persist/ColumnIdExpr.java:138]: table name and column id also need back quoted?

Comment (zhangheihei) [reply fe/fe-core/src/main/java/com/starrocks/persist/ColumnIdExpr.java:138]: tks review.
I don't think it's necessary to add backquotes to tablet.columnId.
In SQL syntax parsing, if use a reserved keyword as an identifier, then user must add backquotes to it, but tablet.columnId is not a reserved keyword.

**Review by wyb — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/persist/ColumnIdExpr.java (hunk 1 above)
Comment (zhangheihei) [reply fe/fe-core/src/main/java/com/starrocks/persist/ColumnIdExpr.java]: Yes, you are right.
The tosql semantics should be guaranteed here

**Review by Seaven — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**